MineSweeper.py

```diff
@@ -0,0 +1,256 @@
+import Tkinter as tk
+import random
+
+import sys
+sys.path.append('~/Library/Python/2.7/site-packages')
+
+from drawingpanel import *
+
+BOARD_HEIGHT = 300
+BOARD_WIDTH = 300
+BOX_SIZE = 20
+
+NUMBER_MINES = ((BOARD_WIDTH / BOX_SIZE - 1)**2)//10 
+
+class Game():
+
+  def __init__(self):
+    self.master = tk.Tk()
+    self.panel = Canvas(self.master, width = BOARD_WIDTH, height = BOARD_HEIGHT + 100, bg = 'white')
+    self.panel.pack()
+
+
+     
+  def game_over(self):
+
+    self.master.unbind("<Button-1>")
+    self.master.unbind("<Button-2>")
+
+    self.b = Button(self.master, text = 'Play Again?', command = self.new_game)
+    self.panel.create_text(BOARD_WIDTH/2, BOARD_HEIGHT + 40, text  = 'Game Over', fill = 'red') 
+    self.b.pack()
+
+  def new_game(self):
+
+    self.master.destroy()
+
+    main()
+
+
+class Board(Game):
+
+  def __init__(self, Game):
+   
+    self.game = Game
+
+    self.__length = BOARD_HEIGHT/BOX_SIZE - 1
+    self.__width = BOARD_WIDTH/BOX_SIZE -1
+
+    self.__tiles = []
+    
+    row = []
+
+    for i in range(self.__length):
+      for j in range(self.__width):
+        new_tile = Tile(i*BOX_SIZE, j*BOX_SIZE, BOX_SIZE)
+        row.append(new_tile)
+
+      self.__tiles.append(row)
+      row = []
+
+
+  def draw(self, panel):
+
+    panel.delete(ALL)
+
+    marked_mines = 0
+
+    for row in self.__tiles:
+      for tile in row:
+        tile.draw(panel)
+      
+        if(tile.marked_mine):
+          marked_mines += 1
+
+    mines = NUMBER_MINES - marked_mines
+
+    bottom_text = 'Mines left: ' + repr(mines)
+    text_fill = 'black'
+
+    panel.create_text(BOARD_WIDTH/2, BOARD_HEIGHT + 20, text  = bottom_text, fill = text_fill)
+
+
+  # assign mines to random locations across the board. 
+  def assign_mines(self):
+
+    mines_to_assign = NUMBER_MINES
+
+    while(mines_to_assign > 0):
+      x_index = random.randint(0, len(self.__tiles)-1)
+      y_index = random.randint(0, len(self.__tiles)-1)
+
+
+      if(self.__tiles[x_index][y_index].is_mine == False):
+        self.__tiles[x_index][y_index].is_mine = True
+        mines_to_assign -= 1
+
+        for i in range(-1, 2):
+          for j in range(-1, 2):
```

Comment:
A couple reasons come to mind:
 - wrapping a chunk of confusing code into a method gives it a label and a name which, hopefully, helps illuminate what the code is doing
 - you could actually do some error checking within that method (like only return valid neighbor coordinates)
 - in general there's an idea about how much complexity you can hold in your head at a time. It's hard to reason about this particular method (for me) because this piece of logic pushes me over (my admittedly small) limit of complexity I can manage. If you just get rid of it and replace it with `for neighbor in self.neighbors` I'm like "oh, nice, iterate over the neighbors".